### PR TITLE
update changelogs of 3.4 and 3.5 for fixing avoid closing a watch with ID 0 incorrectly

### PR DESCRIPTION
**Pull request by kafuu-chino**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

Updating changelogs for https://github.com/etcd-io/etcd/pull/14562 and https://github.com/etcd-io/etcd/pull/14563 .cc @ahrtr  @mitake @spzala

### CODE REVIEW

File: CHANGELOG/CHANGELOG-3.4.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ Previous change logs can be found at [CHANGELOG-3.3](https://github.com/etcd-io/
 ### etcd server
 - Fix [memberID equals zero in corruption alarm](https://github.com/etcd-io/etcd/pull/14530)
 - Fix [auth invalid token and old revision errors in watch](https://github.com/etcd-io/etcd/pull/14548)
+- Fix [avoid closing a watch with ID 0 incorrectly](https://github.com/etcd-io/etcd/pull/14296)
```
Comment (ahrtr) [inline CHANGELOG/CHANGELOG-3.4.md:12]:
```suggestion
- Fix [avoid closing a watch with ID 0 incorrectly](https://github.com/etcd-io/etcd/pull/14562)
```

File: CHANGELOG/CHANGELOG-3.5.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ Previous change logs can be found at [CHANGELOG-3.4](https://github.com/etcd-io/
 
 ### etcd server
 - Fix [auth invalid token and old revision errors in watch](https://github.com/etcd-io/etcd/pull/14547)
+- Fix [avoid closing a watch with ID 0 incorrectly](https://github.com/etcd-io/etcd/pull/14296)
```
Comment (ahrtr) [inline CHANGELOG/CHANGELOG-3.5.md:11]:
```suggestion
- Fix [avoid closing a watch with ID 0 incorrectly](https://github.com/etcd-io/etcd/pull/14563)
```

**Comment by ahrtr** [comment]
Please squash the commits into one commit.

**Comment by kafuu-chino** [comment]
> Please squash the commits into one commit.

Finished. @ahrtr

**Review by mitake — APPROVED** [review]
LGTM, thanks @kafuu-chino ! Defer to @ahrtr

**Review by ahrtr — APPROVED** [review]
LGTM

Thank you @kafuu-chino and @mitake

**Comment by kafuu-chino** [comment]
Thanks very much @ahrtr @mitake 👍

**Pull request merged**